### PR TITLE
Add disabled state styles to Accordion

### PR DESCRIPTION
**Pull request by tmbalagan**
Added styles for the disabled state of accordion items

### CODE REVIEW

**Comment by maneeshp97** [comment]
Hi @tmbalagan,

Thanks for your contribution to Twigs! We appreciate you taking the time to improve our codebase.

We're currently discussing the design changes you've proposed in your pull request with our design team to ensure they align with our overall design guidelines. Specifically, we're reviewing the color palette you've used and determining if any adjustments are needed.

We'll get back to you as soon as possible with our feedback and next steps.

Thanks again for your contribution!

File: packages/react-components/src/accordion/accordion.tsx [hunk 1]
```diff
@@ -49,6 +49,11 @@ const AccordionItem = styled(AccordionPrimitive.Item, {
   '&:focus-within': {
     position: 'relative',
     zIndex: 1
+  },
+  '&[data-disabled]': {
+    background: '$white900',
```
Comment (maneeshp97) [inline packages/react-components/src/accordion/accordion.tsx:54]: @tmbalagan background property can be avoided (opacity alone will be enough) here as the accordion component is expected to work fine in all sorts of backgrounds. Adding white background here will break the accordion ui in screens which have any other color as the background.

File: packages/react-components/src/accordion/accordion.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const StyledTrigger = styled(AccordionPrimitive.Trigger, {
   [`[data-state=open] > & ${StyledChevron}`]: {
     transform: 'rotate(180deg)'
   },
```
Comment (maneeshp97) [inline packages/react-components/src/accordion/accordion.tsx:26]: @tmbalagan  can we also set the cursor on accordion trigger to `'not-allowed'` when accordion item is disabled ?

**Comment by tmbalagan** [comment]
Hi @maneeshp97 

Thank you for reviewing my contribution! Please take a look at the updated code and share any feedback.

File: packages/react-components/src/accordion/stories/accordion.stories.tsx [hunk 3]
```diff
@@ -7,12 +7,16 @@ export default {
...
-    type: 'single'
+    type: 'single',
+    itemDisabled: false
   },
   argTypes: {
     type: {
       control: 'radio',
       options: ['single', 'multiple']
+    },
+    itemDisabled: {
```
Comment (maneeshp97) [inline packages/react-components/src/accordion/stories/accordion.stories.tsx:18]: @tmbalagan  Can we move this to docs (`packages/docs/docs/components/accordion.mdx`) ? We can add the `disabled` prop to the `AccordionItem` prop table which we already have. Calling it `itemDisabled` here while the actual prop is `disabled` can be confusing.

Comment (tmbalagan) [reply packages/react-components/src/accordion/stories/accordion.stories.tsx]: @maneeshp97 updated, Kindly take a look.

**Review by akzhysparrow — APPROVED** [review]

**Comment by maneeshp97** [comment]
@all-contributors please add @tmbalagan for code

**Pull request merged**